presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -0,0 +1,254 @@
+            if (start != Integer.parseInt(name)) {
+                isContinuousSequence = false;
+                break;
+            }
+            start++;
+        }
+
+        if (isContinuousSequence) {
+            return fileNames.get(fileNames.size() - 1);
+        }
+
+        return compressFileNamesUsingRoaringBitmap(fileNames);
+    }
+
+    static List<String> decompressFileNames(String compressedFileNames)
+    {
+        // Check if the compressed fileNames string is a number
+        if (compressedFileNames.matches("\\d+")) {
+            long end = Long.parseLong(compressedFileNames);
+
+            if (end == 0) {
+                return ImmutableList.of("0");
+            }
+
+            return LongStream.range(0, end + 1).mapToObj(String::valueOf).collect(toImmutableList());
+        }
+
+        try {
+            RoaringBitmap roaringBitmap = new RoaringBitmap();
+            ByteBuffer byteBuffer = ByteBuffer.wrap(compressedFileNames.getBytes(ISO_8859_1));
+            roaringBitmap.deserialize(byteBuffer);
+            return Arrays.stream(roaringBitmap.toArray()).mapToObj(Integer::toString).collect(toImmutableList());
+        }
+        catch (IOException e) {
+            throw new PrestoException(MALFORMED_HIVE_FILE_STATISTICS, "Failed de-compressing the file names in manifest");
+        }
+    }
+
+    private static String compressFileNamesUsingRoaringBitmap(List<String> fileNames)
+    {
+        RoaringBitmap roaringBitmap = new RoaringBitmap();
+
+        // Add file names to roaring bitmap
+        fileNames.forEach(name -> roaringBitmap.add(Integer.parseInt(name)));
+
+        // Serialize the compressed data into ByteBuffer
+        ByteBuffer byteBuffer = ByteBuffer.allocate(roaringBitmap.serializedSizeInBytes());
+        roaringBitmap.serialize(byteBuffer);
+        byteBuffer.flip();
+
+        return new String(byteBuffer.array(), ISO_8859_1);
+    }
+
+    public static String compressFileSizes(List<Long> fileSizes)
+    {
+        String fileSizesString = Joiner.on(COMMA).join(fileSizes.stream().map(String::valueOf).collect(toImmutableList()));
+        try {
+            return new String(Zstd.compress(fileSizesString.getBytes(ISO_8859_1), COMPRESSION_LEVEL), ISO_8859_1);
+        }
+        catch (RuntimeException e) {
+            throw new PrestoException(MALFORMED_HIVE_FILE_STATISTICS, "Failed compressing the file sizes for manifest");
+        }
+    }
+
+    public static List<Long> decompressFileSizes(String compressedFileSizes)
+    {
+        try {
+            byte[] compressedBytes = compressedFileSizes.getBytes(ISO_8859_1);
+            long decompressedSize = Zstd.decompressedSize(compressedBytes);
+            String decompressedFileSizes = new String(Zstd.decompress(compressedBytes, (int) decompressedSize), ISO_8859_1);
+            return Arrays.stream(decompressedFileSizes.split(COMMA)).map(Long::valueOf).collect(toImmutableList());
+        }
+        catch (RuntimeException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)